tools/pubsys/src/aws/promote_ssm/mod.rs

```diff
@@ -216,8 +231,86 @@ pub(crate) async fn run(args: &Args, promote_args: &PromoteArgs) -> Result<()> {
     Ok(())
 }
 
+/// Read parameters in given file, add newly promoted parameters, and write combined parameters to
+/// the given file
+async fn write_rendered_parameters(
+    ssm_parameters_output: &PathBuf,
+    set_parameters: &HashMap<SsmKey, String>,
+    source_target_map: HashMap<&String, &String>,
+) -> Result<()> {
+    info!(
+        "Writing promoted SSM parameters to {}",
+        ssm_parameters_output.display()
+    );
+    let parsed_parameters = parse_expected_parameters(&ssm_parameters_output.to_owned())
```

Comment:
Should this be `parse_existing_parameters` or just `parse_parameters`? Seems like the name is a holdover from parsing it in the validation code, but we need the method more generically.

```suggestion
    let parsed_parameters = parse_parameter_file(&ssm_parameters_output.to_owned())
```